crates/subspace-farmer/src/cluster/plotter.rs

```diff
@@ -380,7 +381,11 @@ impl ClusterPlotter {
                     }
                 };
 
-                let (mut sector_sender, sector_receiver) = mpsc::channel(1);
+                // Allow to buffer up to the whole sector in memory to not block plotter on the
+                // other side
+                let (mut sector_sender, sector_receiver) = mpsc::channel(
+                    sector_size(pieces_in_sector) / nats_client.approximate_max_message_size(),
```

Comment:
Is there an edge case here where the sector size is very small, and the message size is very large?

I didn’t do the calculations in detail, but I think that we can get a situation where sector size is around 1 MB, and message size is 2 MB? Which would panic due to a zero-sized channel.

If the minimum number of pieces per sector is high enough, and the upper limit on the message size is low enough, this won’t be a problem, but I’m not familiar enough with this code to know.

Reply:
It is unlikely, but possible. Good catch, thanks!